clang/lib/Driver/Driver.cpp

```diff
@@ -5203,761 +5225,816 @@ class OffloadingActionBuilder final {
       }
     }
 
-    // Return whether to use native bfloat16 library.
-    bool selectBfloatLibs(const ToolChain *TC, bool &useNative) {
-      const char *TargetOpt = nullptr;
-      const char *DeviceOpt = nullptr;
-      bool needLibs = false;
-      for (auto *A : Args) {
-        llvm::Triple *TargetBE = nullptr;
-
-        auto GetTripleIt = [&, this](llvm::StringRef Triple) {
-          llvm::Triple TargetTriple{Triple};
-          auto TripleIt = llvm::find_if(SYCLTripleList, [&](auto &SYCLTriple) {
-            return SYCLTriple == TargetTriple;
-          });
-          return TripleIt != SYCLTripleList.end() ? &*TripleIt : nullptr;
-        };
+    // Performs device specific linking steps for the SYCL based toolchain.
+    // This function is used for both the early AOT flow and the typical
+    // offload device link flow.
+    // The ListIndex input provides an index against the SYCLTargetInfoList
```

Comment:
Same name in comment and code (ListIndex vs LI) will help. Also enumerating each input with description might be a good addition (second request is a nit).

Thanks

Reply:
Will update - thanks